From 35afb7d8968dae29e315196ac3dfab7153e160c5 Mon Sep 17 00:00:00 2001
From: Dan McGee <dan@archlinux.org>
Date: Mon, 1 Sep 2014 14:39:27 -0500
Subject: [PATCH] libmount: don't base GC startup on no-mtab context

When mtab is a symlink to /proc/mounts, libmount uses /run/mount/utab to
store filesystem-specific mount attributes, making this check
unnecessary. Worse, it now breaks nilfs_cleanerd under systemd 216+
since it always passes `-n` to the mount command.

Signed-off-by: Dan McGee <dan@archlinux.org>
---
 sbin/mount/mount_libmount.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/sbin/mount/mount_libmount.c b/sbin/mount/mount_libmount.c
index c518475..b901654 100644
--- a/sbin/mount/mount_libmount.c
+++ b/sbin/mount/mount_libmount.c
@@ -404,12 +404,6 @@ static int nilfs_update_mount_state(struct nilfs_mount_info *mi)
 	rungc = gc_ok && !mi->new_attrs.nogc;
 	old_attrs = (mi->mflags & MS_REMOUNT) ? &mi->old_attrs : NULL;
 
-	if (mnt_context_is_nomtab(cxt)) {
-		if (rungc)
-			printf(_("%s not started\n"), NILFS_CLEANERD_NAME);
-		return 0;
-	}
-
 	if (rungc) {
 		if (mi->new_attrs.pp == ULONG_MAX)
 			mi->new_attrs.pp = mi->old_attrs.pp;
-- 
2.1.0

